lib/tools.js

```diff
@@ -1702,32 +1702,16 @@ function setQualityForInstance(objects, states, namespace, q) {
  * @returns {string}
  */
 function pattern2RegEx(pattern) {
-    pattern = (pattern || '').toString()
-        .replace(/\$/g, '\\$')
-        .replace(/\^/g, '\\^');
+    pattern = (pattern || '').toString();
 
-    if (pattern !== '*') {
-        if (pattern[0] === '*' && pattern[pattern.length - 1] !== '*') {
-            pattern += '$';
-        }
-        if (pattern[0] !== '*' && pattern[pattern.length - 1] === '*') {
-            pattern = '^' + pattern;
-        }
-        if (pattern[0] !== '*' && pattern[pattern.length - 1] !== '*') {
-            pattern = '^' + pattern + '$';
-        }
-    }
+    const startsWithWildcard = pattern[0] === '*';
+    const endsWithWildcard = pattern[pattern.length - 1] === '*';
 
-    pattern = pattern
-        .replace(/\?/g, '\\?')
-        .replace(/\./g, '\\.')
-        .replace(/\(/g, '\\(')
-        .replace(/\)/g, '\\)')
-        .replace(/\[/g, '\\[')
-        .replace(/]/g, '\\]')
-        .replace(/\*/g, '.*');
+    pattern = pattern.replace(/[-/\\^$+?.()|[\]{}]/g, '\\$&')
+        .replace(/\*/g, '.*')
+        .replace(/\?/g, '.'); // supporting ? as single char wild card like redis does
```

Comment:
I think you need to remove the question mark from line 1710 if you replace it here.

Reply:
yes, of course ;-)